std/assembly/rt/tlsf.ts

```diff
@@ -374,7 +374,7 @@ function prepareBlock(root: Root, block: Block, size: usize): void {
 
 /** Adds more memory to the pool. */
 function addMemory(root: Root, start: usize, end: usize): bool {
-  if (DEBUG) assert(start <= end); // must be valid
+  if (DEBUG) assert(start <= end || end == 0); // must be valid
```

Comment:
In a potential future where `usize` might be 64-bit, it seems like this check does nothing. For 32-bit `usize`, the check feels brittle. Would it perhaps make sense to change this to `end64: u64`, then deriving `let end: usize = <usize>end64` from it after the check has been performed?

Reply:
looks great. I will change it.